src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -463,8 +463,9 @@ internal RuntimeMethodHandleInternal GetInterfaceMethodImplementation(RuntimeTyp
         internal static bool IsComObject(RuntimeType type, bool isGenericCOM)
         {
 #if FEATURE_COMINTEROP
+            // We need to check the type handle values - not the instances - to determine if the runtime type is a ComOjbect.
```

Comment:
New instances are created for different COM classes (so that COM objects of different CLSIDs have different types. For example, if you have:
```c#
Type t1 = Type.GetTypeFromCLSID(<clsid>);
Type t2 = Type.GetTypeFromCLSID(<clsid>);
Type t3 = Type.GetTypeFromCLSID(<different_clsid>);
```
They would be `__ComObject`, but `t1` and `t3` would be different instances such that `t1 == t2` but `t1 != t3`.